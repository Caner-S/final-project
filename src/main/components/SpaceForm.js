import React from 'react';
import {getUnavailableBookings} from "../dao/BookingDao";
import {addSpace} from "../dao/SpaceDao";

class SpaceForm extends React.Component {
    constructor() {
        super();
        this.state = {
            number: null,
            capacity: null,
            type: null,
            exclusive: null,
            section: null,
            floor: null
         };
    }

    updateInput = e => {
        this.setState({
            [e.target.name]: e.target.value
        });

    }

    addOfficeSpace = e => {
        e.preventDefault();
        addSpace( this.state.number, this.state.capacity, this.state.type, this.state.exclusive, this.state.section, this.state.floor);

        this.setState({
            number: '',
            capacity: '',
            type: '',
            exclusive: '',
            section: '',
            floor: ''
    });
    };

    render() {
        return (
            <div>
                <h4>New Office space</h4>
            <form onSubmit={this.addOfficeSpace}>
                <input
                    type="number"
                    name="number"
                    placeholder="Number"
                    onChange={this.updateInput}
                    value={this.state.number}
                        />
                <input
                    type="number"
                    name="capacity"
                    placeholder="Capacity"
                        onChange={this.updateInput}
                    value={this.state.capacity}
                        />
                <select
                    name="type"
                    placeholder="Type"
                        onChange={this.updateInput}
                    value={this.state.type}
                        >
                    <option value="Desk">Desk</option>
                    <option value="Meeting Room">Meeting Room</option>
                    <option value="Workspace">Workspace</option>
                </select>
                <input
                    type="text"
                    name="exclusive"
                    placeholder="Exclusive"
                        onChange={this.updateInput}
                    value={this.state.exclusive}
                        />
                <input
                    type="number"
                    name="section"
                    placeholder="Section"
                    onChange={this.updateInput}
                    value={this.state.section}
                />
                <input
                    type="number"
                    name="floor"
                    placeholder="Floor"
                        onChange={this.updateInput}
                    value={this.state.floor}
                        />
                <button type="submit">Submit</button>
            </form>
            </div>
        );
    }
}
export default SpaceForm;