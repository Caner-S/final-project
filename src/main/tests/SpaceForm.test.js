import React from 'react';
import renderer from 'react-test-renderer';

import SpaceForm from "../components/SpaceForm";

it('renders correctly when there are no items', () => {
    const tree = renderer.create(<SpaceForm />).toJSON();
    expect(tree).toMatchSnapshot();
});