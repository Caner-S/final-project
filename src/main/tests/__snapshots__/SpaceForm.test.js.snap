// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div>
  <h4>
    New Office space
  </h4>
  <form
    onSubmit={[Function]}
  >
    <input
      name="number"
      onChange={[Function]}
      placeholder="Number"
      type="number"
      value={null}
    />
    <input
      name="capacity"
      onChange={[Function]}
      placeholder="Capacity"
      type="number"
      value={null}
    />
    <select
      name="type"
      onChange={[Function]}
      placeholder="Type"
      value={null}
    >
      <option
        value="Desk"
      >
        Desk
      </option>
      <option
        value="Meeting Room"
      >
        Meeting Room
      </option>
      <option
        value="Workspace"
      >
        Workspace
      </option>
    </select>
    <input
      name="exclusive"
      onChange={[Function]}
      placeholder="Exclusive"
      type="text"
      value={null}
    />
    <input
      name="section"
      onChange={[Function]}
      placeholder="Section"
      type="number"
      value={null}
    />
    <input
      name="floor"
      onChange={[Function]}
      placeholder="Floor"
      type="number"
      value={null}
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
